File: Module2/newR/MCPi_0a.R
# MonteCarlo Pi 
# MCPi_0.R
# M. Saum 8/31/2017
rm(list=ls())

set.seed(843)

NTrials <- c(10, 100, 1000, 10000, 100000, 1000000)
Ncases <- length(NTrials)
PiEst <- numeric(Ncases)

for( i in 1:Ncases) {
  x <- runif(NTrials[i])
  y <- runif(NTrials[i])
  Rcount <- sum(x^2 + y^2 <= 1)
  PiEst[i] <- 4*Rcount/NTrials[i]
}

sink("MCPi_0_R.txt")
print(cbind(NTrials,PiEst))
sink()

pdf("MCPi_0_R.pdf")
plot(NTrials,PiEst)
dev.off()

File: Module2/newOctave/MCPi_0a.m
% MonteCarlo Pi
% MCPi_0.m
% M. Saum 8/31/2017
clear;

rand("seed",843);

NTrials = [10; 100; 1000; 10000; 100000; 1000000];
Ncases = numel(NTrials);
PiEst = zeros(Ncases,1);

for i =1:Ncases 
  x = rand(NTrials(i), 1);
  y = rand(NTrials(i), 1);
  Rcount = sum(x.^2 + y.^2 <= 1); 
  PiEst(i) = 4*Rcount/NTrials(i);
end

dlmwrite("MCPi_0_m.txt",[NTrials,PiEst]);

hf = figure();
plot(NTrials',PiEst');
print(hf,"MCPi_0_m.pdf");
